autoload/neural/buffer.vim

```diff
@@ -0,0 +1,28 @@
+scriptencoding utf-8
+
+function! neural#buffer#CreateBuffer() abort
+    " TODO: Customise buffer options
+    vertical new
+    setlocal wrap linebreak
+
+    file Neural Buffer
+    setlocal buftype=nofile
+    setlocal bufhidden=hide
+    setlocal noswapfile
+    setlocal filetype=neuralbuf
```

Comment:
Above:

```vim
let l:original_filetype = &filetype
```

Then here:

```suggestion
    let &filetype = l:original_filetype . '.neuralbuf'
```

Then the buffer has the same type as the other thing, we can check the results with ALE, ALEFix it, and all other nice things. We may want to add an option to turn this off.